### PR TITLE
Fix for 3632

### PR DESCRIPTION
**Pull request by ChamaraPhilipsuom**
Removing Oauth, Token and UserInfo endpoint from ui.

### CODE REVIEW

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2AuthenticationConstant.java [hunk 1]
```diff
@@ -22,7 +22,7 @@ private GoogleOAuth2AuthenticationConstant() {
     }
 
     public static final String GOOGLE_OAUTH_ENDPOINT = "GoogleOauthEndpoint";
-    public static final String GOOGLE_TOKEN_ENDPOINT = "GooglrTokenEndpoint";
+    public static final String GOOGLE_TOKEN_ENDPOINT = "GoogleTokenEndpoint";
     public static final String GOOGLE_USERINFO_ENDPOINT = "GoogleuserInfoEndpoint";
     public static final String GOOGLE_CONNECTOR_FRIENDLY_NAME = "Google OAuth2/OpenID Connect";
     public static final String GOOGLE_CONNECTOR_NAME = "GoogleOAUth2OpenIDAuthenticator";
```
Comment (johannnallathamby) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2AuthenticationConstant.java]: The constants and their values don't seem consistent.
public static final String GOOGLE_AUTHZ_ENDPOINT = "GoogleAuthzEndpoint";
public static final String GOOGLE_TOKEN_ENDPOINT = "GoogleTokenEndpoint";
public static final String GOOGLE_USERINFO_ENDPOINT = "GoogleUserInfoEndpoint";
     public static final String GOOGLE_CONNECTOR_FRIENDLY_NAME = "Google OAuth2/OpenID Connect";
     public static final String GOOGLE_CONNECTOR_NAME = "GoogleOIDCAuthenticator";

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2Authenticator.java [hunk 2]
```diff
@@ -57,33 +57,85 @@
...
+        }
+    }
+
+    /**
+     * Get UserInfo Endpoint
+     *
+     * @return userInfoEndpoint
+     */
+
+    private String getUserInfoEndpoint() {
```
Comment (johannnallathamby) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2Authenticator.java]: Shall we pass Map<String, String> authenticatorProperties here as well to be consistent.

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java [hunk 3]
```diff
@@ -59,11 +62,59 @@
...
+        if (StringUtils.isBlank(this.userInfoEndpoint)) {
+            this.userInfoEndpoint = IdentityApplicationConstants.WINDOWS_LIVE_USERINFO_URL;
+        }
+    }
+
+    /**
+     *
+     * @return userInfoEndpoint
+     */
+    private String getUserInfoEndpoint() {
```
Comment (johannnallathamby) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/live/WindowsLiveOAuth2Authenticator.java]: Shall we pass Map authenticatorProperties here as well to be consistent.

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java [hunk 4]
```diff
@@ -122,8 +193,8 @@ protected void processAuthenticationResponse(HttpServletRequest request,
                     authenticatorProperties.get(FacebookAuthenticatorConstants.CLIENT_SECRET);
             String userInfoFields = authenticatorProperties.get(FacebookAuthenticatorConstants.USER_INFO_FIELDS);
 
-            String tokenEndPoint = authenticatorProperties.get(FacebookAuthenticatorConstants.FB_TOKEN_URL);
-            String fbauthUserInfoUrl = authenticatorProperties.get(FacebookAuthenticatorConstants.FB_USER_INFO_URL);
+            String tokenEndPoint = getTokenEndpoint();
+            String fbauthUserInfoUrl = getUserInfoEndpoint();
```
Comment (pulasthi7) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: variable name is not in camel case.

Comment (ChamaraPhilipsuom) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: Fixed at #1682

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java [hunk 5]
```diff
@@ -55,6 +56,76 @@
 
     private static final long serialVersionUID = -4844100162196896194L;
     private static final Log log = LogFactory.getLog(FacebookAuthenticator.class);
+    private String tokenEndpoint;
+    private String oAuthEndpoint;
+    private String userInfoEndpoint;
+
+
+    /**
+     * initiate tokenEndpoint
```
Comment (prabathabey) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: Use proper linguistics for comments. Start the sentence with a capital letter.

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java [hunk 6]
```diff
@@ -55,6 +56,76 @@
...
+    private void initTokenEndpoint() {
+        this.tokenEndpoint = getAuthenticatorConfig().getParameterMap().get(FacebookAuthenticatorConstants
+                .FB_TOKEN_URL);
+        if (StringUtils.isBlank(this.tokenEndpoint)) {
+            this.tokenEndpoint = IdentityApplicationConstants.FB_TOKEN_URL;
+        }
+    }
+
+    /**
+     * initiate authorization server endpoint
```
Comment (prabathabey) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: Same goes here.

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java [hunk 7]
```diff
@@ -55,6 +56,76 @@
...
+    private void initOAuthEndpoint() {
+        this.oAuthEndpoint = getAuthenticatorConfig().getParameterMap().get(FacebookAuthenticatorConstants
+                .FB_AUTHZ_URL);
+        if (StringUtils.isBlank(this.oAuthEndpoint)) {
+            this.oAuthEndpoint = IdentityApplicationConstants.FB_AUTHZ_URL;
+        }
+    }
+
+    /**
+     * initiate userInfoEndpoint
```
Comment (prabathabey) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: Here as well.

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java [hunk 8]
```diff
@@ -55,6 +56,76 @@
...
+    private void initUserInfoEndPoint() {
+        this.userInfoEndpoint = getAuthenticatorConfig().getParameterMap().get(FacebookAuthenticatorConstants
+                .FB_USER_INFO_URL);
+        if (StringUtils.isBlank(this.userInfoEndpoint)) {
+            this.userInfoEndpoint = IdentityApplicationConstants.FB_USER_INFO_URL;
+        }
+    }
+
+    /**
+     * get the tokenEndpoint.
```
Comment (prabathabey) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: Here too.

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2Authenticator.java [hunk 9]
```diff
@@ -57,33 +57,85 @@
...
 
     private static final long serialVersionUID = -4154255583070524018L;
+    private String tokenEndpoint;
+    private String oAuthEndpoint;
+    private String userInfoEndpoint;
 
     private static Log log = LogFactory.getLog(GoogleOAuth2Authenticator.class);
 
     /**
+     * initiate tokenEndpoint
```
Comment (prabathabey) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2Authenticator.java]: Use proper linguistics for comments. Start respective sentences with a capital letter in all similar instances.

Comment (ChamaraPhilipsuom) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/google/GoogleOAuth2Authenticator.java]: Fixed at #1682

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java (hunk 8 above)
Comment (ChamaraPhilipsuom) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: Fixed at #1682

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java (hunk 7 above)
Comment (ChamaraPhilipsuom) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: Fixed at #1682

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java (hunk 6 above)
Comment (ChamaraPhilipsuom) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: Fixed at #1682

File: components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java (hunk 5 above)
Comment (ChamaraPhilipsuom) [inline components/application-authenticators/org.wso2.carbon.identity.application.authenticator.social/src/main/java/org/wso2/carbon/identity/application/authenticator/social/facebook/FacebookAuthenticator.java]: Fixed at #1682

**Pull request merged**